api/apps/api/src/modules/scenarios/scenarios.service.ts

```diff
@@ -319,49 +476,86 @@ export class ScenariosService {
   async getOneSolution(
     scenarioId: string,
     runId: string,
+    userId: string,
     _fetchSpecification: FetchSpecification,
-  ) {
+  ): Promise<Either<typeof forbiddenError, ScenariosOutputResultsApiEntity>> {
     await this.assertScenario(scenarioId);
-    // TODO permissions guard
-    // TODO runId is part of scenarioId
```

Comment:
Has this been handled?

Reply:
Permissions yes, but I overlooked that I removed the latter part so I added it again.